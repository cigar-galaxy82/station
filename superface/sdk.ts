import { createTypedClient } from "@superfaceai/one-sdk";

import { communicationSendEmail } from "./types/communication/send-email";
import { communicationSendMessage } from './types/communication/send-message';

export { CommunicationSendEmailProfile } from "./types/communication/send-email";
export const typeDefinitions = {
    ...communicationSendEmail,
    ...communicationSendMessage
};
export const SuperfaceClient = createTypedClient(typeDefinitions);
