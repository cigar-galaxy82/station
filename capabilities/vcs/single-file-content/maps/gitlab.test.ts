import { ok, SuperfaceClient } from '@superfaceai/sdk';

describe('vcs/single-file-content/gitlab', () => {
  it('performs correctly', async () => {
    const client = new SuperfaceClient();
    const profile = await client.getProfile('vcs/single-file-content');
    const useCase = profile.getUseCase('singleFileContent');
    const provider = await client.getProvider('gitlab');

    await expect(
      useCase.perform(
        {
          owner: 'Jakub-Vacek',
          repo: 'empty-test',
          path: 'README.md',
          ref: 'master',
        },
        { provider }
      )
    ).resolves.toEqual(
      ok({
        content: expect.any(String),
        encoding: 'base64',
        size: expect.any(Number),
      })
    );
  });
});
