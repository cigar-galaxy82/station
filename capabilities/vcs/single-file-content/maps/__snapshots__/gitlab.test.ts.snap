// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vcs/single-file-content/gitlab SingleFileContent should map error 1`] = `
Err {
  "error": "HTTPError: Expected HTTP error
Properites: {
  \\"message\\": \\"404 Commit Not Found\\",
  \\"statusCode\\": 404
}
AST Path: definitions[0].statements[3].responseHandlers[0].statements[1]
Original Map Location: Line 19, column 7",
}
`;

exports[`vcs/single-file-content/gitlab SingleFileContent should perform successfully 1`] = `
Ok {
  "value": Object {
    "content": "MQojIGVtcHR5LXRlc3QKdGhpcyBpcyBlbXB0eSByZXBvCg==",
    "encoding": "base64",
    "size": 34,
  },
}
`;
